### ISSUE TITLE
Fails on StrictRedis

### ISSUE THREAD

**Issue by Glueon** [gabrielfalcao/HTTPretty#271]
I am using the version available from pypi and it still has a bug when used with redis.

If I disable unmocked connections I get:

```
httpretty.errors.UnmockedError: No mocking was registered, and real connections are not allowed (httpretty.allow_net_connect = False)
```

Otherwise it gives an error:

```
redis.exceptions.ConnectionError: Error 9 while writing to socket. Bad file descriptor.
```

I tried calling `ping()` call before mocking and it helped. Now there is case where it does not work. If that is patched in version which is on git - I can't install it as it fails with:

```
 raise RuntimeError("Couldn't find the `requirements/use.txt' file :(")
```

**Comment by dustinfarris** [comment]
I see this if i register more than one uri

**Comment by gabrielfalcao** [comment]
closed at f0bb2bdc1f5e2771e0cb854f57014e7d0fcc6b71
